test/ruby/test_settracefunc.rb

```diff
@@ -594,7 +594,7 @@ def trace_by_set_trace_func
 
     eval <<-EOF.gsub(/^.*?: /, ""), nil, 'xyzzy'
     1: set_trace_func(lambda{|event, file, line, id, binding, klass|
-    2:   events << [event, line, file, klass, id, binding.eval('self'), binding.eval("_local_var")] if file == 'xyzzy'
+    2:   events << [event, line, file, klass, id, (binding.eval('self') if binding), (binding.eval("_local_var") if binding)] if file == 'xyzzy'
```

Comment:
```suggestion
    2:   events << [event, line, file, klass, id, binding&.eval('self'), binding&.eval("_local_var")] if file == 'xyzzy'
```

Nitpick: Safe navigation for different aesthetics if you want.